include/ttmlir/Dialect/TTNN/Analysis/L1InterleavedPolicy.h

```diff
@@ -23,6 +44,58 @@ class L1InterleavedPolicy : public MemoryLayoutAnalysisPolicy {
                                    schedule, usableL1CacheSize) {}
 
   void run() final;
+
+private:
+  // Check if the op is analyzable. Op is analyzable if it has at least one
+  // legal layout.
+  bool isAnalyzable(Operation *op);
+
+  // Fetch op's DRAM layout from legalLayouts.
+  bool hasDRAMLayout(Operation *op);
+  TTNNLayoutAttr getDRAMLayout(Operation *op);
+
+  // Fetch op's L1 Interleaved layout from legalLayouts.
+  bool hasL1InterleavedLayout(Operation *op);
+  TTNNLayoutAttr getL1InterleavedLayout(Operation *op);
+
+  size_t getAvailableL1CacheSize() const {
+    // Figure out this const based on exec data, but will be replaced
+    // with API.
+    //
+    constexpr float tensorL1UsageCap = 0.75;
+    return tensorL1UsageCap * usableL1CacheSize;
+  }
+
+  // Get optimal op's config base on the outputs of its operands.
+  Config getOptimalConfig(Operation *baseOp,
+                          llvm::DenseMap<Operation *, L1Usage> &opsL1Usage);
```

Comment:
Are opsL1Usage operands of baseOp? If so, maybe a better name would be operandL1Usage?

Reply:
That is not true. The `opsL1Usage` contains all the information about the execution of the `baseOp`. So, `L1Usage`s of both `baseOp` and its operands is stored in the `opsL1Usage` object. The reason for this is that the algorithm for determining the optimal l1 usage of the op does not make distinction between 'baseOp' and its operands. It just creates the configuration of L1 layouts so that the L1 memory is maximally used.